docs/debugging-templates.md

```diff
@@ -0,0 +1,64 @@
+# Debugging Templates
+
+Module templates can be debugged visually through use of a `TemplateDebugView`
+that renders a single module with a series of test cases.
+
+Consider an example module that renders a simple hyperlink. This module
+requires a target URL and also accepts optional link text. A simple Jinja
+template for this module might look like this:
+
+```html
+<a href="{{ value.url }}">
+    {{ value.text | default( value.url, true ) }}
+</a>
+```
+
+## Defining template test cases
+
+Next, define test cases for the module that cover all supported input
+configurations. For this simple link module, there are only a few useful
+test cases, but more complicated modules might have many more.
+
+Test cases should be defined as a Python dict, where the key is a string name
+of the test case and the value is a dict that will be passed to the module
+template.
+
+```py
+# myapp/template_debug.py
```

Comment:
Does it matter that you have `myapp/template_debug.py` here but the actual file path is `myapp/view/template_debug.py`?

Reply:
So the actual view itself lives in the v1 app, because it has to live somewhere. This is the single Django view used to render any templates form anywhere. Individual apps (including v1) also need a place to define the test data used when debugging templates, and I propose to use a convention of myapp.template_debug for this.